### PR TITLE
ntimes execution support bugfix

### PR DESCRIPTION
**Pull request by palashmarantas**
# Issue Link
https://github.com/LambdaTest/test-at-scale/issues/97

# Description
Calling runners process N times to run test N times

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested with push and pull events
- [x] Tested with N > 1 values

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/testexecutionservice/testexecution.go [hunk 1]
```diff
@@ -88,51 +88,64 @@ func (tes *testExecutionService) Run(ctx context.Context,
...
-
 	// FIXME:  commenting this out as we will need to rework on coverage logic after test parallelization
 	// if collectCoverage {
 	// 	if err := tes.createCoverageManifest(tasConfig, coverageDir, removedfiles, executeAll); err != nil {
 	// 		tes.logger.Errorf("failed to create manifest file %v", err)
 	// 		return nil, err
 	// 	}
 	// }
-	return executionResults, errC
+	azureWriter.Close()
```
Comment (VikrantKS) [inline pkg/testexecutionservice/testexecution.go:143]: this call is deferred in line 49

File: pkg/testexecutionservice/testexecution.go [hunk 2]
```diff
@@ -88,51 +88,64 @@ func (tes *testExecutionService) Run(ctx context.Context,
...
+	if uploadErr := <-errChan; uploadErr != nil {
+		tes.logger.Errorf("failed to upload logs for test execution, error: %v", uploadErr)
+		return nil, uploadErr
+	}
+	return executionResults, nil
```
Comment (saurabh-prakash) [inline pkg/testexecutionservice/testexecution.go:148]: why is this added?

Comment (palashmarantas) [reply pkg/testexecutionservice/testexecution.go:148]: to return execution results . We are using this to find task fail or pass

Comment (palashmarantas) [reply pkg/testexecutionservice/testexecution.go:148]: Line 144 to 148 we already exist in code..

Comment (palashmarantas) [reply pkg/testexecutionservice/testexecution.go]: removed these lines. these are added by mistake during conflict resolve process.

File: pkg/testexecutionservice/testexecution.go (hunk 1 above)
Comment (palashmarantas) [reply pkg/testexecutionservice/testexecution.go]: removed

File: pkg/testexecutionservice/testexecution.go (hunk 2 above)
Comment (palashmarantas) [reply pkg/testexecutionservice/testexecution.go]: in case any exceptions in tests will get test failure only.

**Review by VikrantKS — DISMISSED** [review]

**Review by saurabh-prakash — APPROVED** [review]

**Review by utkarsh-lambdatest — APPROVED** [review]
